linumpy/io/thorlabs.py

```diff
@@ -0,0 +1,421 @@
+        cropped_data = data[:, :, index1:index2]
+        self.size_z = cropped_data.shape[2]
+
+        return cropped_data
+
+    def _load_raw_data(self, file) -> np.ndarray:
+        """
+        Load the raw data from the specified file and return it as a NumPy array.
+
+        Parameters:
+            file (str): File path in the compressed data.
+
+        Returns:
+            np.ndarray: Raw complex data array.
+        """
+        with self.compressed_data.open(file) as f:
+            raw_data = np.frombuffer(f.read(), dtype=np.complex64).reshape(
+                (self.size_x, self.size_y, self.size_z), order="C"
+            )
+        return raw_data
+
+    def _preprocess_data(
+        self,
+        data: np.ndarray,
+    ) -> np.ndarray:
+        """
+        Preprocess the data, including cropping, stacking, and converting to magnitude.
+
+        Parameters:
+            data (np.ndarray): Input complex data array.
+
+        Returns:
+            np.ndarray: Preprocessed data array.
+        """
+        # Perform cropping
+        data = self._crop_z(
+            data,
+            index1=self.config.crop_first_index,
+            index2=self.config.crop_second_index,
+        )
+        # Perform stacking
+        data = self._stack_tiles_vertically(data)
+        # Adjust the size_y to be divisible by ascan_averaging_value. Necessary for stacking.
+        data = data[:, : data.shape[1] - (data.shape[1] % self.ascan_averaging_value), :]
+        self.size_y = data.shape[1]
+
+        # Return complex or magnitude data
+        return data if self.config.return_complex else np.abs(data).astype(np.float64)
+
+    def load_and_process(self, file: str) -> np.ndarray:
+        """
+        Load raw data from the file and preprocess it.
+
+        Parameters:
+            file (str): File path in the compressed data.
+
+        Returns:
+            np.ndarray: Fully processed data array.
+        """
+        raw_data = self._load_raw_data(file)
+        processed_data = self._preprocess_data(raw_data)
+        return processed_data
+
+    @staticmethod
+    def extract_positions_from_scan(scan_file_path: str = None):
+        """
+        Extracts the raw and index x, y positions from the .scan file.
+
+        Parameters:
+        - scan_file_path: Path to the .scan file.
+        - number_of_tiles: Number of tiles in the scan.
+
+        Returns:
+        - tuple: A tuple containing two lists - index positions and raw positions.
+        """
+        raw_positions = []
+
+        if scan_file_path:
+            with open(file=scan_file_path, mode="r", encoding="utf-8") as file:
+                lines = file.readlines()
+
+                # Find the start of the positions section
+                positions_section = False
+                for line in lines:
+                    line = line.strip()
+
+                    # Mark the start of the positions section
+                    if line == "------Positions------":
```

Comment:
I don't know the input, but wouldn't `if "Positions" in line:` be a good enough condition?